Sources/MapboxDirections/RouteLeg.swift

```diff
@@ -447,7 +447,6 @@ public extension Array where Element == RouteLeg {
 private extension Array {
     mutating func replace(subrange: PartialRangeFrom<Int>, with newElements: Array?) {
         guard let newElements = newElements else { return }
-        precondition(subrange.lowerBound < newElements.count)
```

Comment:
This precondition actually does not make sense. It checks that we are replacing a subrange, starting from an element index, which is lower than number of new elements we are inserting. These are unrelated values. 
As an example:
We have 5 numeric congestions array - `[0, 1, 2, 3, 4]` and we want to update 2 last items to values `[5, 6]`. But since `subrange.lowerBound (3) > newElements.count (2)` this check will fail the update.

Reply:
If directions endpoint returns invalid number of updated annotations - the resulting annotations array may change it's size. It should not lead to a crash, but annotations display may loose some tailing information.
As a safeguard we can check that `subrange.lowerBound..<self.count` contains at least `newElements.count` items...